pyro/distributions/diag_normal_mixture_shared_cov.py

```diff
@@ -18,92 +18,99 @@ class MixtureOfDiagNormalsSharedCovariance(TorchDistribution):
-    derivative in your research.
+    derivative in your research. Note that this distribution does not support
+    dimension D = 1.
 
     [1] Pathwise Derivatives for Multivariate Distributions, Martin Jankowiak &
     Theofanis Karaletsos. arXiv:1806.01856
 
     :param torch.Tensor locs: K x D mean matrix
-    :param torch.Tensor scale: shared D-dimensional scale vector
-    :param torch.Tensor logits: K-dimensional vector of softmax logits
+    :param torch.Tensor coord_scale: shared D-dimensional scale vector
+    :param torch.Tensor component_logits: K-dimensional vector of softmax logits
     """
     has_rsample = True
-    arg_constraints = {"locs": constraints.real, "scale": constraints.positive,
-                       "logits": constraints.real}
+    arg_constraints = {"locs": constraints.real, "coord_scale": constraints.positive,
+                       "component_logits": constraints.real}
 
-    def __init__(self, locs, scale, logits):
+    def __init__(self, locs, coord_scale, component_logits):
         self.batch_mode = (locs.dim() > 2)
         assert(self.batch_mode or locs.dim() == 2), \
             "The locs parameter in MixtureOfDiagNormals should be K x D dimensional (or ... x B x K x D in batch mode)"
         if not self.batch_mode:
-            assert(scale.dim() == 1), "The scale parameter in MixtureOfDiagNormals should be D dimensional"
-            assert(logits.dim() == 1), "The logits parameter in MixtureOfDiagNormals should be K dimensional"
-            assert(logits.size(0) == locs.size(0))
+            assert(coord_scale.dim() == 1), "The coord_scale parameter in MixtureOfDiagNormals should be D dimensional"
+            assert(component_logits.dim() == 1), \
+                "The component_logits parameter in MixtureOfDiagNormals should be K dimensional"
+            assert(component_logits.size(0) == locs.size(0))
             batch_shape = ()
         else:
-            assert(scale.dim() > 1), "The scale parameter in MixtureOfDiagNormals should be ... x B x D dimensional"
-            assert(logits.dim() > 1), "The logits parameter in MixtureOfDiagNormals should be ... x B x K dimensional"
-            assert(logits.size(-1) == locs.size(-2))
+            assert(coord_scale.dim() > 1), \
+                "The coord_scale parameter in MixtureOfDiagNormals should be ... x B x D dimensional"
+            assert(component_logits.dim() > 1), \
+                "The component_logits parameter in MixtureOfDiagNormals should be ... x B x K dimensional"
+            assert(component_logits.size(-1) == locs.size(-2))
             batch_shape = tuple(locs.shape[:-2])
         self.locs = locs
-        self.scale = scale
-        self.logits = logits
+        self.coord_scale = coord_scale
+        self.component_logits = component_logits
         self.dim = locs.size(-1)
-        self.categorical = Categorical(logits=logits)
+        assert (self.dim > 1), "This distribution does not support dimension D = 1"
```

Comment:
Can you raise a `NotImplementedError` here?
```py
if self.dim < 2:
    raise NotImplementedError('This distribution does not support D = 1')
```
Then you might get a mug.

Reply:
done. @jpchen pleaaaasssse go get me a mug!?